java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java

```diff
@@ -1568,6 +1574,52 @@ public String downloadSystemId(User loggedInUser, Integer sid) throws FaultExcep
         return maps;
     }
 
+    /**
+     * List current package locks status.
+     * @param loggedInUser The current user
+     * @param sid System ID
+     * @return Returns an array of maps representing the packages locked on a system
+
+
+     * @xmlrpc.doc List current package locks status.
+     * @xmlrpc.param #session_key()
+     * @xmlrpc.param #param("string", "system_id")
+     * @xmlrpc.returntype
+     *      #array_begin()
+     *          #struct_begin("package")
+     *                 #prop_desc("int", "package_id", "PackageID, -1 if package is locked but not available in
+     *                 subscribed channels")
+     *                 #prop("string", "name")
+     *                 #prop("string", "epoch")
+     *                 #prop("string", "version")
+     *                 #prop("string", "release")
+     *                 #prop_desc("string", "arch", "architecture label")
+     *                 #prop_desc("string", "pending status", "return only if there is a pending locking")
+     *          #struct_end()
+     *      #array_end()
+     */
+    public List<Map<String, Object>>  listLocksStatus(User loggedInUser,  Integer sid) {
```

Comment:
I would choose a different name for this function.

We can also "lock a system". Please could missunderstand this. 
Maybe listPackagesLockStatus() ?